lib/elixir_analyzer/test_suite/dancing_dots.ex

```diff
@@ -0,0 +1,94 @@
+defmodule ElixirAnalyzer.TestSuite.DancingDots do
+  @moduledoc """
+  This is an exercise analyzer extension module for the concept exercise rpg-character-sheet
+  """
+
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.Source
+
+  use ElixirAnalyzer.ExerciseTest
+
+  feature "DancingDots.Flicker shouldn't reimplement init/1" do
+    type :essential
+    find :none
+    comment Constants.dancing_dots_do_not_reimplement_init()
+
+    form do
+      defmodule DancingDots.Flicker do
+        _block_includes do
+          def init(_ignore), do: _ignore
+        end
+      end
+    end
+  end
+
+  check_source "uses @impl DancingDots.Animation" do
+    type :actionable
+    comment Constants.dancing_dots_annotate_impl_animation()
+
+    check(%Source{code_ast: code_ast}) do
+      {_, %{aliases: aliases}} =
+        Macro.prewalk(
+          code_ast,
+          %{aliases: %{}},
+          &find_aliases/2
+        )
+
+      {_, %{defs_without_impls: defs_without_impls}} =
+        Macro.prewalk(
+          code_ast,
+          %{defs_without_impls: [], impl?: false, skip?: false, aliases: aliases},
+          &find_defs_and_impls/2
+        )
+
+      defs_without_impls == []
+    end
+  end
+
+  @def_ops [:def, :defp, :defmacro, :defmacrop, :defguard, :defguardp]
+  @callbacks_in_this_exercise [:init, :handle_frame]
+  @behaviour_module [:DancingDots, :Animation]
+  defp find_aliases(node, acc) do
+    acc =
+      case node do
+        {:alias, _, [{:__aliases__, _, module_name}]} ->
```

Comment:
It might be simpler at this stage to only care about aliases of `@behaviour_module`, this way `acc` could simplify to a list (which could also be initialized as `[@behaviour_module]` to simplify the check later).
Unless you intended to reuse the function, but then this is not the right place...

Reply:
Sure: [`f02db59` (#293)](https://github.com/exercism/elixir-analyzer/pull/293/commits/f02db5922c3595826adbaf6e187ec378ce238e77)